src/api/app/models/kiwi/image/xml_builder.rb

```diff
@@ -14,12 +14,56 @@ def build
         doc = update_packages(doc)
         doc = update_repositories(doc)
         doc = update_description(doc)
+        doc = update_preferences(doc)
 
         Nokogiri::XML(doc.to_xml, &:noblanks).to_xml(indent: kiwi_indentation(doc))
       end
 
       private
 
+      def update_preferences(document)
+        return document if @image.preference_type.blank?
+
+        # <preferences> block does not exist
+        if document.xpath('image/preferences').any?
+          # <preferences> and <type> blocks exist already
+          if document.xpath('image/preferences/type').any?
```

Comment:
Also the type is AFAIK mandatory so we should already validate that while importing, no?